client/src/components/History/CurrentHistory/HistoryPanel.vue

```diff
@@ -147,21 +149,47 @@ export default {
             return `${this.history.id}&${this.queryString}`;
         },
         queryString() {
-            return new SearchParams(this.params).historyContentQueryString;
+            const deleted = this.params.showDeleted ? "True" : "False";
+            const visible = this.params.showHidden ? "False" : "True";
+            return `q=deleted&q=visible&qv=${deleted}&qv=${visible}`;
         },
     },
     methods: {
+        getState(item) {
+            if (item.job_state_summary) {
+                for (const key of ["error", "failed", "paused", "upload", "running"]) {
+                    if (item.job_state_summary[key] > 0) {
+                        return key;
+                    }
+                }
+                return "ok";
+            } else {
+                return item.state || item.populated_state;
```

Comment:
item.state and item.populated_state are different concepts, I wouldn't mix them together.
Collections can be pre-created, in which case they already have members, or you can create a collection and then add members. In the first case populated_state is `ok` form the start, in the latter the state is `new`, then becomes `ok`. And if could be `failed` if something goes wrong when mapping over jobs for instance.